### PR TITLE
Remove TombstoneDocSupplier from EngineConfig

### PR DESCRIPTION
**Pull request by dnhatn**
With #2251, we can create delete and noop tombstones directly.

Relates #72251

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java [hunk 1]
```diff
@@ -354,27 +350,6 @@ public LongSupplier getPrimaryTermSupplier() {
         return primaryTermSupplier;
     }
 
-    /**
-     * A supplier supplies tombstone documents which will be used in soft-update methods.
-     * The returned document consists only _uid, _seqno, _term and _version fields; other metadata fields are excluded.
-     */
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java:360]: Maybe it's worth moving this javadoc to the `ParsedDocument` tombstone factory methods?

**Review by romseygeek — APPROVED** [review]
LGTM!

File: server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java:360]: +1. Adjusted in https://github.com/elastic/elasticsearch/pull/72593/commits/56cb22dd88f078fc188faf27a34dcab5e4884c37.

**Comment by dnhatn** [comment]
Thanks Alan!

**Pull request merged**